test/inst_update_partition_auto_test.rb

```diff
@@ -0,0 +1,207 @@
+      stub_subject(subject)
+    end
+
+    context "when installation is restarting" do
+      let(:restarting) { true }
+
+      it "loads data from data dump file if present" do
+        expect(subject).to receive(:data_stored?).and_return(true)
+        expect(subject).to receive(:load_data)
+
+        expect(subject.main).to eql(:cancel)
+      end
+    end
+
+    context "when root partition is mounted" do
+      before do
+        allow(Yast::RootPart).to receive(:Mounted).and_return(true)
+        allow(Yast::RootPart).to receive(:UnmountPartitions)
+      end
+
+      it "detachs update" do
+        expect(Yast::Update).to receive(:Detach)
+
+        subject.main
+      end
+
+      it "unmounts all the root partitions" do
+        expect(Yast::RootPart).to receive(:UnmountPartitions).with(false)
+
+        subject.main
+      end
+    end
+
+    it "obtains the current target system" do
+      expect(subject).to receive(:current_target_system).once
+
+      subject.main
+    end
+
+    context "when a target system can't be obtained" do
+      before do
+        allow(subject).to receive(:current_target_system).and_return(nil)
+      end
+
+      it "shows the partition dialog" do
+        expect(subject).to receive(:RootPartitionDialog)
+
+        subject.main
+      end
+    end
+
+    context "when a target system can be obtained" do
+      before do
+        allow(subject).to receive(:current_target_system).and_return("/dev/whatever")
+      end
+
+      it "sets the RootPart.selectedRootPartition with its value" do
+        expect(Yast::RootPart).to receive(:selectedRootPartition=).with("/dev/whatever")
+
+        subject.main
+      end
+
+      it "tries to mount the target" do
+        expect(Yast::RootPart).to receive(:mount_target)
+
+        subject.main
+      end
+
+      context "when the target system is not mounted" do
+        before do
+          allow(Yast::RootPart).to receive(:mount_target).and_return(false)
+        end
+
+        it "reports and Error" do
+          expect(Yast::Report).to receive(:Error).once
+
+          subject.main
+        end
+
+        it "unmount all the partitions" do
+          expect(subject).to receive(:UmountMountedPartition).once
+
+          subject.main
+        end
+
+        it "shows the partition dialog" do
+          expect(subject).to receive(:RootPartitionDialog)
+
+          subject.main
+        end
+      end
+
+      context "when the target system is mounted successfully" do
+        before do
+          allow(Yast::RootPart).to receive(:mount_target).and_return(true)
+          allow(subject).to receive(:store_data)
+        end
+
+        context "when it detects a incomplete installation" do
+          before do
+            allow(Yast::RootPart).to receive(:IncompleteInstallationDetected).and_return(true)
+          end
+
+          it "reports an error" do
+            expect(Yast::Report).to receive(:Error).once
```

Comment:
Maybe we could check the message (just to be sure you're reporting the right error). But be aware of i18n :)